### PR TITLE
Fix memory issues with certain Animated GIFs

### PR DESCRIPTION
**Pull request by paulcbetts**
This PR fixes an issue where certain GIFs would immediately cause all available memory to be allocated, which Seems Bad

/cc @niik @shiftkey

### CODE REVIEW

File: src/Update/Update-Mono.csproj [hunk 1]
```diff
@@ -70,16 +70,11 @@
...
-    <Reference Include="WpfAnimatedGif, Version=1.4.14.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\WpfAnimatedGif.1.4.14\lib\net\WpfAnimatedGif.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
   </ItemGroup>
   <ItemGroup>
     <Compile Include="..\SolutionAssemblyInfo.cs">
       <Link>Properties\SolutionAssemblyInfo.cs</Link>
     </Compile>
-    <Compile Include="AnimatedGifWindow.cs" />
```
Comment (shiftkey) [inline src/Update/Update-Mono.csproj]: So no Mono support for this? Or did this not work at all before?

File: src/Update/TaskbarManager.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        /// <summary>
+        /// Indicates whether this feature is supported on the current platform.
+        /// </summary>
+        public static bool IsPlatformSupported
+        {
+            get
+            {
+                // We need Windows 7 onwards ...
+                return true;
```
Comment (shiftkey) [inline src/Update/TaskbarManager.cs]: Worth actually checking here? Or is the "no parents" approach Good Enough?

File: src/Update/AnimatedGifWindowWinForms.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+            pictureBox = new PictureBox();
+            this.Controls.Add(pictureBox);
+
+            if (File.Exists(source)) {
+                pictureBox.ImageLocation = source;
+            }
+
+            this.WindowState = FormWindowState.Minimized;
+            Action size = () => { pictureBox.Width = this.Width; pictureBox.Height = this.Height; pictureBox.Left = 0; pictureBox.Top = 0; };
```
Comment (shiftkey) [inline src/Update/AnimatedGifWindowWinForms.cs]: omg (lack of) indenting

File: src/Update/Update-Mono.csproj (hunk 1 above)
Comment (paulcbetts) [inline src/Update/Update-Mono.csproj]: Mono is only for building packages (aka `--releasify`), no changes here

File: src/Update/TaskbarManager.cs (hunk 2 above)
Comment (paulcbetts) [inline src/Update/TaskbarManager.cs]: .NET 4.5. is >= Win7 only anyways, right?

File: src/Update/AnimatedGifWindowWinForms.cs (hunk 3 above)
Comment (paulcbetts) [inline src/Update/AnimatedGifWindowWinForms.cs]: This probably needs some cleanup, I kinda yolo'd out this code

File: src/Update/TaskbarManager.cs (hunk 2 above)
Comment (shiftkey) [inline src/Update/TaskbarManager.cs]: [Vista SP2](https://docs.microsoft.com/en-us/dotnet/framework/get-started/system-requirements)

Comment (paulcbetts) [inline src/Update/TaskbarManager.cs]: 👎 👎 👎

**Pull request merged**